app/services/course_search_service.rb

```diff
@@ -36,7 +23,34 @@ def call
     scope = scope.within(filter[:radius], origin: origin) if locations_filter?
     scope = scope.with_funding_types(funding_types) if funding_types.any?
 
-    scope.distinct
+    # The 'where' scope will remove duplicates
+    # An outer query is required in the event the provider name is present.
+    # This prevents 'PG::InvalidColumnReference: ERROR: for SELECT DISTINCT, ORDER BY expressions must appear in select list'
+    outer_scope = Course.where(id: scope.pluck(:id))
```

Comment:
Would it worth using `scope.select(:id)` here instead of `pluck`? I'm thinking Postgresql might be able to optimise the sub-query if we keep it all in one query.

```suggestion
    outer_scope = Course.where(id: scope.select(:id))
```

Reply:
Done

Reply:
Just for future reference, it looks like `pluck` can be faster than `select` for small (around 30 or less) result-sets. I tested a few scenarios:

**Searching for all Universtity of Warwick (W20) courses on Find**
courses returned: `16`
average select db time increase: `+95.26ms`
average select total time increase: `+157.80ms`

**Searching for all University College London (U80) courses on Find**
courses returned: `28`
average select db request time increase: `+49.28ms`
average select total request time increase: `-30.3ms`

**Searching for all Primary courses on Find**
courses returned: `1,580`
average select db time increase: `-103.7ms`
average select total time increase: `-400.40ms`

(lower times mean `select` is faster than `pluck`)

I think the gains from speeding up larger results is probably worth more than the losses from slowing down smaller ones, but we'd probably need to do a proper study of the distribution of results we end up returning is, and the speed-up/slow-down we'd get, would be needed.